### PR TITLE
Support using suffix of key for directory in addition to the prefix.

### PR DESCRIPTION
**Pull request by kevina**
Closes #6

### CODE REVIEW

File: flatfs.go [hunk 1]
```diff
@@ -22,43 +22,47 @@ import (
...
+		return str[len(str)-suffixLen:]
+	}
+}
 
 func (fs *Datastore) encode(key datastore.Key) (dir, file string) {
 	noslash := key.String()[1:]
-	prefix := (noslash + padding)[:fs.prefixLen]
-	dir = path.Join(fs.path, prefix)
+	dir = path.Join(fs.path, fs.getDir(noslash))
 	file = path.Join(dir, noslash+extension)
```
Comment (kevina) [inline flatfs.go]: no, we need to return dir

File: flatfs.go [hunk 2]
```diff
@@ -22,43 +22,47 @@ import (
...
 	sync bool
 }
 
 var _ datastore.Datastore = (*Datastore)(nil)
 
-func New(path string, prefixLen int, sync bool) (*Datastore, error) {
-	if prefixLen <= 0 || prefixLen > maxPrefixLen {
-		return nil, ErrBadPrefixLen
-	}
+func New(path string, getDir func(string) string, sync bool) (*Datastore, error) {
```
Comment (kevina) [inline flatfs.go]: Done.

File: flatfs.go [hunk 3]
```diff
@@ -22,43 +22,49 @@ import (
...
-	ErrBadPrefixLen = errors.New("bad prefix length")
+	extension = ".data"
 )
 
 type Datastore struct {
 	path string
-	// length of the dir splay prefix
-	prefixLen int
+
+	getDir func(string) string
```
Comment (whyrusleeping) [inline flatfs.go]: `getDir` should be of type `ShardFunc`

Comment (kevina) [inline flatfs.go]: Done.  Sorry.

**Pull request merged**